### PR TITLE
New Package:py-sierrapy@0.3.0

### PR DESCRIPTION
**Pull request by dorton21**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-sierrapy/package.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class PySierrapy(PythonPackage):
+    """A Client of HIVdb Sierra GraphQL Webservice"""
+
+    homepage = "https://github.com/hivdb/sierra-client/tree/master/python"
+    pypi     = "sierrapy/sierrapy-0.3.0.tar.gz"
+
+    maintainers = ['dorton21']
+
+    version('0.3.0', sha256='82474b3815d79d16a480b0cc70b9f7075430ff4990f33306c880b240a3141b6e')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sierrapy/package.py:18]: Missing build dep on setuptools, missing a few other dependencies (note that these all need to be exact versions, not version ranges):
```python
certifi==2020.4.5.1
chardet==3.0.4
click==7.1.2
gql==0.4.0
graphql-core==2.3.1
idna==2.9
promise==2.3
requests==2.23.0
rx==1.6.1
six==1.14.0
tqdm==4.46.0
urllib3==1.25.9
voluptuous==0.11.7
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-sierrapy/package.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    depends_on('py-chardet@3.0.4', type=('build', 'run'))
+    depends_on('py-click@7.1.2', type=('build', 'run'))
+    depends_on('py-gql@0.4.0', type=('build', 'run'))
+    depends_on('py-graphql-core@2.3.2', type=('build', 'run'))
+    depends_on('py-idna@2.9', type=('build', 'run'))
+    depends_on('py-rx@1.6.1', type=('build', 'run'))
+    depends_on('py-promise@2.3', type=('build', 'run'))
+    depends_on('py-requests@2.23.0', type=('build', 'run'))
+    depends_on('py-six@1.14.0', type=('build', 'run'))
+    depends_on('py-tqdm@4.51.0', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sierrapy/package.py]:
```suggestion
    depends_on('py-tqdm@4.46.0', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-sierrapy/package.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    depends_on('py-voluptuous@0.11.7', type=('build', 'run'))
+    depends_on('py-urllib3@1.25.10', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sierrapy/package.py]:
```suggestion
    depends_on('py-urllib3@1.25.9', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by dorton21** [comment]
Depends on updates to: #23927, #23928
Depends on new package: #23929

**Review by adamjstewart — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-sierrapy/package.py (hunk 1 above)
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/py-sierrapy/package.py:18]: @adamjstewart We don't have these versions in spack, @dorton21 just took your remands to get it merged.

(@tldahlgren JFYI ) Neither you, nor @dorton21 could have ever built the package with these versions, or the versions would have had to be submitted as well. Pinning this large number of versions must have been clear to be not satisfiable just by pure luck.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-sierrapy/package.py:18]: Yes, we really need better build tests through GitLab CI in the future. We get far too many PRs for me to manually build all of them, so it's not really possible to catch these kinds of mistakes at review time.

**Pull request merged**